include/networkit/centrality/GroupClosenessGrowShrink.hpp

```diff
@@ -0,0 +1,665 @@
+class GroupClosenessGrowShrinkGeneral final : public Algorithm {
+
+public:
+    /**
+     * Finds a group of nodes with high group closeness centrality. This is the Grow-Shrink
+     * algorithm presented in Angriman et al. "Local Search for Group Closeness Maximization on Big
+     * Graphs" IEEE BigData 2019. The algorithm takes as input a graph and an arbitrary group of
+     * nodes, and improves the group closeness of the given group by performing vertex exchanges.
+     *
+     * @param G A connected undirected graph.
+     * @param first, last A range that contains the initial group of nodes.
+     * @param extended Set this parameter to true for the Extended Grow-Shrink algorithm (i.e.,
+     * vertex exchanges are not restricted to only neighbors of the group).
+     * @param insertions Number of consecutive node insertions and removal per iteration. Let this
+     * parameter to zero to use Diameter(G)/sqrt(k) nodes (where k is the size of the group).
+     */
+    template <class Iter>
+    GroupClosenessGrowShrinkGeneral(const Graph &graph, Iter first, Iter last,
+                                    bool extended = false, count insertions = 0)
+        : G(&graph), group(first, last), extended(extended), insertions(insertions),
+          heap(CompareDistance(distance)), heap_(CompareDistance(distance_)) {
+
+        if (G->isDirected()) {
+            std::runtime_error("Error, this algorithm does not support directed graphs.");
+        }
+
+        if (group.empty()) {
+            throw std::runtime_error("Error, empty group.");
+        }
+    }
+
+    // Easier constructor to wrap in Cython.
+    GroupClosenessGrowShrinkGeneral(const Graph &graph, const std::vector<node> &group,
+                                    bool extended = false, count insertions = 0)
+        : GroupClosenessGrowShrinkGeneral(graph, group.begin(), group.end(), extended, insertions) {
+    }
+
+    /**
+     * Returns the computed group.
+     */
+    std::vector<node> groupMaxCloseness() const {
+        assureFinished();
+        std::vector<node> result;
+        result.reserve(group.size());
+        for (const auto &entry : idxMap) {
+            result.push_back(entry.first);
+        }
+
+        return result;
+    }
+
+    /**
+     * Returns the total number of iterations performed by the algorithm.
+     */
+    count numberOfIterations() const {
+        assureFinished();
+        return totalSwaps;
+    }
+
+    // Maximum number of iterations allowed.
+    count maxIterations{100};
+
+    // Random seed used for the estimation of the transitive closure of the DAG.
+    uint64_t randomSeed{1};
+
+    /**
+     * Runs the algorithm.
+     */
+    void run() override {
+        init();
+
+        if (!insertions) {
+            Diameter diam(*G, DiameterAlgo::estimatedRange, 0.1);
+            diam.run();
+            DEBUG("Diameter upper bound = ", diam.getDiameter().second);
+            insertions = std::max(1., .5
```

Comment:
Just to be sure: Is the truncation from `double` to `int` intended?

Reply:
Yes, as we want the number of insertions to be an integer >= 1.